client/my-sites/jetpack-search/main-jetpack.tsx

```diff
@@ -40,18 +41,24 @@ export default function JetpackSearchMainJetpack( { siteId }: Props ): ReactElem
 	// Have we loaded the necessary purchases and modules? If not, show the placeholder.
 	const modules = useSelector( ( state ) => getJetpackModules( state, siteId ) );
 
-	const isRequestingModules =
-		useSelector( ( state ) => isFetchingJetpackModules( state, siteId ) ) || ! modules;
+	const isRequestingModules = useSelector( ( state ) => isFetchingJetpackModules( state, siteId ) );
 
 	// On Jetpack sites, we need to check if the search module is active, rather than checking settings.
 	const isJetpackSearchModuleActive = useSelector( ( state ) =>
 		isJetpackModuleActive( state, siteId, 'search' )
 	);
 
-	if ( isRequestingPurchases || isRequestingModules ) {
+	// isRequestingModules is null if a request hasn't been triggered yet
+	const isLoading = isRequestingPurchases || isRequestingModules || isRequestingModules === null;
```

Comment:
```suggestion
	const isLoading = isRequestingPurchases || isRequestingModules || !!isRequestingModules;
```
Would the use of `!!` better here just in case it's value is `false` somehow?

Reply:
Not in this case - I discovered the states work like this:

isRequestingModules === null - I have not requested any modules yet
isRequestingModules === true - I am requesting modules now
isRequestingModules === false - I have finished requested modules and they should be in Redux

... so `null` and `false` are different outcomes.

Reply:
@bluefuton sweet TIL. Thanks for the explanation then all good to go 👍